packages/editor/src/lib/editor/tools/SelectTool/children/Idle.ts

```diff
@@ -375,7 +375,7 @@ export class Idle extends StateNode {
 		this.parent.transition('editing_shape', info)
 	}
 
-	private createTextShapeAtPoint(info: TLClickEventInfo) {
+	handleDoubleClickOnCanvas(info: TLClickEventInfo) {
```

Comment:
Note that this PR also opens up the `handleDoubleClickOnCanvas`, which you could overwrite if you like to prevent double clicking on canvas from creating a text shape.